frontend/src/pages/answer/Answer.jsx

```diff
@@ -192,13 +216,98 @@ export default function Answer() {
     event.target.value = '';
   }
 
+  /**
+   * Save an uploaded answer to Robokache
+   */
+  async function saveAnswer() {
+    let accessToken;
+    if (isAuthenticated) {
+      try {
+        accessToken = await getAccessTokenSilently();
+      } catch (err) {
+        displayAlert('error', `Failed to validate user. Error: ${err}`);
+      }
+    } else {
+      return displayAlert('warning', 'You need to be signed in to save an answer.');
+    }
+    let response = await API.cache.createAnswer(defaultAnswer, accessToken);
+    if (response.status === 'error') {
+      return displayAlert('error', 'Failed to create answer.');
+    }
+    const answerId = response.id;
+    response = await API.cache.setAnswerData(answerId, { message: answerStore.message }, accessToken);
+    if (response.status === 'error') {
+      return displayAlert('error', 'Failed to save answer.');
+    }
+    return displayAlert('success', 'Your answer has been saved!');
+  }
+
+  /**
+   * Delete an answer from Robokache
+   *
+   * Also check if the associated question has other answers
+   * and set `hasAnswers` to false if it doesn't
+   */
+  async function deleteAnswer() {
+    let accessToken;
+    if (isAuthenticated) {
+      try {
+        accessToken = await getAccessTokenSilently();
+      } catch (err) {
+        displayAlert('error', `Failed to validate user. Error: ${err}`);
+      }
+    } else {
+      return displayAlert('warning', 'You need to be signed in to delete an answer.');
+    }
+    // get associated question
+    let response = await API.cache.getAnswer(answer_id, accessToken);
+    if (response.status === 'error') {
+      return displayAlert('error', `Failed to fetch answer: ${response.message}`);
+    }
+    const questionId = response.parent;
+    // delete answer from Robokache
+    response = await API.cache.deleteAnswer(answer_id, accessToken);
+    if (response.status === 'error') {
+      return displayAlert('error', `Failed to delete answer: ${response.message}`);
+    }
+    history.push('/answer');
+    let alertType = 'success';
+    let alertText = 'Your answer has been deleted!';
+    // check if question has other answers
+    if (questionId) {
+      response = await API.cache.getAnswersByQuestion(questionId, accessToken);
+      if (response.status === 'error') {
+        alertType = 'error';
+        alertText += ` However, failed to get sibling answers: ${response.message}`;
+      } else if (!response.length) {
+        // if question has no answers, set hasAnswers to false
+        response = await API.cache.getQuestion(questionId, accessToken);
+        if (response.status === 'error') {
+          alertType = 'error';
+          alertText += ` However, failed to get associated question data: ${response.message}`;
+        } else {
+          response.metadata.hasAnswers = false;
+          response = await API.cache.updateQuestion(response, accessToken);
```

Comment:
It seems odd that this is the UI's job. Also, would it make more sense for hasAnswers to be `false` by default, and to set it to `true` only after we get answers?

Reply:
This function is working on deleting an answer. Right here, we've just determined that we deleted all answers. So the question had `hasAnswers=true`, now we need to set it to `false` since we just deleted the last answer. I think the reason the UI has to do this was decided in Robokache, because we wanted it to be ignorant of any metadata and data it is storing. As generic document storage, it shouldn't require that metadata have a specific property.

Reply:
I think we should rethink this `hasAnswers` business, but let's roll with it for now.